pkg/setup/setup.go

```diff
@@ -0,0 +1,168 @@
+		if strings.Contains(os.Args[0], "/_test/") ||
+			strings.HasSuffix(os.Args[0], ".test") ||
+			flag.Lookup("test.v") != nil ||
+			flag.Lookup("test.run") != nil {
+			env = EnvironmentTest
+		} else {
+			log.Debug().Msgf("Defaulting to local environment: \n os.Args: %v", os.Args)
+			env = EnvironmentLocal
+		}
+	}
+	return env
+}
+
+func getBacalhauRepoPath() (string, error) {
+	// BACALHAU_DIR has the highest precedence, if its set, we return it
+	repoDir := os.Getenv("BACALHAU_DIR")
+	if repoDir != "" {
+		log.Debug().Str("repo", repoDir).Msg("using BACALHAU_DIR as bacalhau repo")
+		return repoDir, nil
+	}
+	// next precedence is station configuration
+
+	//If FIL_WALLET_ADDRESS is set, assumes that ROOT_DIR is the config dir for Station
+	//and not a generic environment variable set by the user
+	if _, set := os.LookupEnv("FIL_WALLET_ADDRESS"); set {
+		repoDir = os.Getenv("ROOT_DIR")
+		if repoDir != "" {
+			log.Debug().Str("repo", repoDir).Msg("using station ROOT_DIR as bacalhau repo")
+			return repoDir, nil
+		}
+	}
+
+	// next is the repo flag
+
+	if repoDir = viper.GetString("repo"); repoDir != "" {
+		log.Debug().Str("repo", repoDir).Msg("using --repo flag value as bacalhau repo")
+		return repoDir, nil
+	}
+
+	// last is the default, the home directory
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", fmt.Errorf("failed to get user home dir for bacalhau repo: %w", err)
+	}
+	repoDir = filepath.Join(home, ".bacalhau")
+	log.Info().Str("repo", repoDir).Msg("using $HOME for bacalhau repo")
+	return repoDir, nil
+}
+
+// SetupBacalhauRepo ensures that a bacalhau repo and config exist and are initalized.
+func SetupBacalhauRepo(repoDir string) (string, error) {
+	env := getEnvironment()
+
+	var bacalhauConfig types.BacalhauConfig
+	switch env {
+	case EnvironmentProd:
+		bacalhauConfig = configenv.Production
+	case EnvironmentStaging:
+		bacalhauConfig = configenv.Staging
+	case EnvironmentDev:
+		bacalhauConfig = configenv.Development
+	case EnvironmentTest:
+		bacalhauConfig = configenv.Testing
+	case EnvironmentLocal:
+		bacalhauConfig = configenv.Local
+	default:
+		// this would indicate an error in the above logic of `getEnvironment()`
+		bacalhauConfig = configenv.Local
+	}
+
+	if repoDir == "" {
+		var err error
+		repoDir, err = getBacalhauRepoPath()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	fsRepo, err := setupRepo(repoDir, &bacalhauConfig)
+	if err != nil {
+		return "", err
+	}
+	return fsRepo.Path()
+}
+
+func setupRepo(path string, cfg *types.BacalhauConfig) (*repo.FsRepo, error) {
+	fsRepo, err := repo.NewFS(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create repo: %w", err)
+	}
+	if exists, err := fsRepo.Exists(); err != nil {
+		return nil, fmt.Errorf("failed to check if repo exists: %w", err)
+	} else if !exists {
+		if err := fsRepo.Init(cfg); err != nil {
+			return nil, fmt.Errorf("failed to initalize repo: %w", err)
+		}
```

Comment:
Is it worth addong a check for an empty dir and also creating a repo if the dir exists but is empty? I initially created the repo dir and then was suprised when Bac couldn't create a repo there:

```
% BACALHAU_DIR=/tmp/repo bin/darwin_arm64/bacalhau docker run ubuntu date
failed to initialize bacalhau repo at '/tmp/repo': failed to open repo: stat /tmp/repo/config.yaml: no such file or directory
% BACALHAU_DIR=/tmp/repo bin/darwin_arm64/bacalhau --repo=/tmp/repo docker run ubuntu date
failed to initialize bacalhau repo at '/tmp/repo': failed to open repo: stat /tmp/repo/config.yaml: no such file or directory
% bin/darwin_arm64/bacalhau --repo=/tmp/repo docker run ubuntu date 
failed to initialize bacalhau repo at '/tmp/repo': failed to open repo: stat /tmp/repo/config.yaml: no such file or directory
```

Or alternatively a clear error.

Reply:
I have modified the code to use the directory if its present and empty. This will work now without error.